From b5f7bca86354fe85b709db8d8fa1908a506ca042 Mon Sep 17 00:00:00 2001
From: Heinrich Schuchardt <xypron.glpk@gmx.de>
Date: Sat, 28 Oct 2017 14:18:22 +0200
Subject: [PATCH 1/1] efi_loader: efi_bootmgr: do not make hidden assignments

Assignments should not be made in the middle of nowhere.

I deliberately check _DEBUG here and not DEBUG because this is what
debug() does. Often I have been setting _DEBUG in the middle of C
files to get only relevant debug messages.

Signed-off-by: Heinrich Schuchardt <xypron.glpk@gmx.de>
---
 lib/efi_loader/efi_bootmgr.c | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/lib/efi_loader/efi_bootmgr.c b/lib/efi_loader/efi_bootmgr.c
index 857d88a879..a55f210f0b 100644
--- a/lib/efi_loader/efi_bootmgr.c
+++ b/lib/efi_loader/efi_bootmgr.c
@@ -120,11 +120,13 @@ static void *try_load_entry(uint16_t n, struct efi_device_path **device_path,
 
 	if (lo.attributes & LOAD_OPTION_ACTIVE) {
 		efi_status_t ret;
-		u16 *str = NULL;
+#if _DEBUG
+		u16 *str = efi_dp_str(lo.file_path);
 
 		debug("%s: trying to load \"%ls\" from: %ls\n", __func__,
-		      lo.label, (str = efi_dp_str(lo.file_path)));
+		      lo.label, str);
 		efi_free_pool(str);
+#endif /* _DEBUG */
 
 		ret = efi_load_image_from_path(lo.file_path, &image);
 
-- 
2.14.2

